src/components/CardElement.js

```diff
@@ -0,0 +1,133 @@
+import { cardSelectors } from "../utils/constants.js";
+import { cardSettings } from "../utils/constants.js";
+
+export default class CardElement {
+  constructor({
+    name,
+    link,
+    isLiked,
+    id,
+    imageClickManager,
+    onDeleteAPIManager,
+    onLikeAPIManager,
+    deleteConfirmationManager,
+  }) {
+    this.id = id;
+    this._cardTemplate = document.querySelector(cardSelectors.template);
+    this._cardElement = this._cardTemplate.content
+      .querySelector(".card")
+      .cloneNode(true);
+    this._cardImage = this._cardElement.querySelector(cardSelectors.image);
+    this._likeButton = this._cardElement.querySelector(
+      cardSelectors.likeButton,
+    );
+    this._deleteButton = this._cardElement.querySelector(
+      cardSelectors.deleteButton,
+    );
+    this._titleElement = this._cardElement.querySelector(
+      cardSelectors.cardTitle,
+    );
+
+    this._cardUnlikedImage = this._cardElement.querySelector(
+      cardSelectors.unlikedImage,
+    );
+    this._cardLikedImage = this._cardElement.querySelector(
+      cardSelectors.likedImage,
+    );
+
+    this._onDeleteAPIManager = onDeleteAPIManager;
+
+    this._onLikeAPIManager = onLikeAPIManager;
+    this.deleteConfirmationManager = deleteConfirmationManager;
+    this._imageClickManager = imageClickManager;
+    this.setTitle(name);
+    this.setCardImage(link);
+    this.setCardImageAlt(name);
+    this._setEventListeners();
+    this._setLike(isLiked);
+  }
+
+  _setEventListeners() {
+    this._cardImage.addEventListener("click", () => this.onImageClick());
+    this._likeButton.addEventListener("click", () => this.onLikeClick());
+    this._deleteButton.addEventListener("click", () => this._onDeleteClick());
+  }
+
+  setTitle(title) {
+    if (typeof title === "string") {
+      this.titleText = title;
+      this._titleElement.textContent = this.titleText;
+    } else {
+      console.log(`New card title is not a string. Setting to default.`);
+      this.titleText = cardSettings.defaultTitle;
+      this._titleElement.textContent = this.titleText;
+    }
+  }
+
+  setCardImage(imageLink) {
+    this.imageLink = imageLink;
+    //how can I make sure the image is loaded before we setAttribute?
+    //feels not in scope atm but I really, really want to fix it.
+    // console.log(imageLink);
+    this._cardImage.setAttribute("src", this.imageLink);
+  }
+
+  setCardImageAlt(altText) {
+    this.imageAltText = altText ? altText : this.titleText;
+    this._cardImage.setAttribute("alt", this.imageAltText);
+  }
+
+  _setLike(bool) {
+    bool ? this.activateLikeButton() : this.deactivateLikeButton();
+  }
+
+  onLikeClick() {
+    if (this.isLikeActive()) {
+      this._onLikeAPIManager(this.id, false).then((result) => {
+        if (result) this.deactivateLikeButton();
+      });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**